### PR TITLE
Add logic for SSID permission

### LINKED ISSUE #8240: Set-DbaNetworkCertificate - Assign permissions to Service SID
### Summarize Functionality

Currently permissions are being granted to the service account however a better approach i feel is to grant to the Sql Service SID as documented in technical details section.

Granting to the service SID will prevent loss of permissions should the service account change.

### Is there a command that is similiar or close to what you are looking for?

No

### Technical Details

https://docs.microsoft.com/en-us/sql/relational-databases/security/using-service-sids-to-grant-permissions-to-services-in-sql-server?view=sql-server-ver15

https://www.sqlskills.com/blogs/bobb/about-sql-servers-usage-of-service-sids/

### PR DESCRIPTION
**Pull request by ACALVETT**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [X] New feature (non-breaking change, adds functionality, fixes #8240  )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

### Purpose
Adds the service SSID to the certificates permissions when calling Set-DbaNetworkPermission

### CODE REVIEW

**Comment by potatoqualitee** [comment]
this looks reasonable to me @andreasjordan how about you?

**Comment by potatoqualitee** [comment]
if andreas doesnt have time, @ACALVETT can you please submit screenshots? also sorry for the delay! it's one of those months

**Comment by ACALVETT** [comment]
No problem, will get screenshots

**Comment by ACALVETT** [comment]
image
Screenshot showing perms granted to SSID

**Comment by andreasjordan** [comment]
I have seen this but not enought time to have a deeper look. I would like to fully understand that thing. Will try to invest some time next week.

**Comment by niphlod** [comment]
jm2c: the logic is correct (that's what I'm used to do "manually"), I cannot vouch for the correctness of the implementation. 

recap: you're running SQL with a specific service user. If that user has not at least read access to the private key, the SQL Service cannot start. Instead of figuring out what service user needs the specific permission, this uses the virtual account to add the ACL once. This means that the day you set your sql server to run with ANOTHER user, you don't need to remember to add the new user to the ACL, nor need to remember to remove the old one from the ACL, and it all "just works".
In many ways this should be already handled by SMO's capabilities of setting the service account (the same one used in the SQL Server configuration mmc) but, unfortunately, it does not.
The same deal is achieved (if you care to look at it) to the data, log and backup folders: sql needs permissions to them, and the setup slams NT SERVICE\MSSQLSERVER at setup time instead of using the service account configured to run the service. I guess they did it for the exact same purpose of this PR, which is "set and forget", instead of adding the new one and removing the old one.

File: functions/Set-DbaNetworkCertificate.ps1 [hunk 1]
```diff
@@ -160,8 +160,16 @@ function Set-DbaNetworkCertificate {
...
 
+                #Grant permissions to the Service SID
+                $sqlSSID = "NT SERVICE\MSSQLSERVER"
+                if ($instanceName -ne "MSSQLSERVER") {
+                    $sqlSSID = "NT SERVICE\MSSQL$" + $instanceName
+                }
+
                 $permission = $serviceAccount, "Read", "Allow"
                 $accessRule = New-Object -TypeName System.Security.AccessControl.FileSystemAccessRule -ArgumentList $permission
+                $permission += $sqlSSID, "Read", "Allow"
```
Comment (andreasjordan) [inline functions/Set-DbaNetworkCertificate.ps1:171]: I think this must be "=" instead if "+=".

**Comment by potatoqualitee** [comment]
(updating branch to rerun tests that failed due to  short github outage)

**Comment by potatoqualitee** [comment]
Lookin good 💯  Thank you all!

**Pull request merged**